### PR TITLE
follow-up(doc, fix): support for grpcroute ratelimitfilter

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

fix: change the logic when dealing with authen and ratelimit filters

refactor: move `getRateLimitFilters` func into filters.go for consistency

doc: add doc for the usage of grpcroute ratelimitfilter

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

follow-up of #1762

### CODE REVIEW

File: docs/latest/user/rate-limit.md [hunk 1]
```diff
@@ -13,14 +13,15 @@ i.e. if the data plane has 2 replicas of Envoy running, and the rate limit is 10
...
+can be linked to a [HTTPRoute][] or [GRPCRoute][] resource using an [ExtensionRef][] filter.
 
 ## Prerequisites
 
 ### Install Envoy Gateway
 
-* Follow the steps from the [Quickstart Guide](quickstart.md) to install Envoy Gateway and the example manifest.
-Before proceeding, you should be able to query the example backend using HTTP.
+* Follow the steps from the [Quickstart Guide](quickstart.md) to install Envoy Gateway and the HTTPRoute example manifest,
+or install the GRPCRoute example from [GRPC Routing](grpc-routing.md).
```
Comment (arkodg) [inline docs/latest/user/rate-limit.md:23]: maybe move this under `### GRPCRoute` because thats not the common case

File: internal/provider/kubernetes/routes.go [hunk 2]
```diff
@@ -177,16 +177,16 @@ func (r *gatewayAPIReconciler) processGRPCRoutes(ctx context.Context, gatewayNam
 					authFilter, ok := resourceMap.authenFilters[key]
```
Comment (arkodg) [inline internal/provider/kubernetes/routes.go:177]: this logic doesnt look right, shouldnt it look more like https://github.com/envoyproxy/gateway/blob/a731995bb3b56366a98eb0b256cc3bec3726e46f/internal/provider/kubernetes/routes.go#L355 which is based on `switch`.

tldr - its not an actual error if the filter is not a AuthenticationFilter and we are trying to fetch something that doesnt exist

Comment (shawnh2) [reply internal/provider/kubernetes/routes.go:177]: fixed

File: docs/latest/user/rate-limit.md (hunk 1 above)
Comment (shawnh2) [reply docs/latest/user/rate-limit.md:23]: updated

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by Xunzhuo — APPROVED** [review]
